sencha-workspace/packages/slate-cbl/src/field/attachments/Field.js

```diff
@@ -174,12 +175,31 @@ Ext.define('Slate.cbl.field.attachments.Field', {
         return normalValue;
     },
 
+    filterValue: function(value) {
+        var me = this,
+            filteredValue = [],
+            length = value ? value.length : 0,
+            i = 0;
+
+        for (; i < length; i++) {
+            if (value[i].Status.toLowerCase() !== 'removed') {
+                filteredValue.push(value[i]);
+            }
+        }
+
+        return filteredValue;
+    },
+
     setValue: function(value) {
         var me = this,
             listCt = me.listCt,
             valueItemsMap = me.valueItemsMap = {},
-            length = value ? value.length : 0,
-            i = 0, itemValue, itemClass, Attachment, attachmentItem;
+            i = 0, length, itemValue, itemClass, Attachment, attachmentItem;
+
+        // filter out items marked as 'removed' if applicable
+        if (me.displayRemoved === false) {
+            value = me.filterValue(value);
+        }
```

Comment:
make sure to use getter when accessing `me.getDisplayeRemoved()`, and if `filterValue` is just going to be called `filterValue` we should make it a standard part of the lifecycle and leave it to that method to read options

```suggestion
        // remove any filtered items
        value = me.filterValue(value);
```